src/Adapter/MSTestAdapter.PlatformServices/Execution/TypeCache.cs

```diff
@@ -669,7 +669,7 @@ private MethodInfo GetMethodInfoForTestMethod(TestMethod testMethod, TestClassIn
 
         MethodInfo? testMethodInfo = testMethod.HasManagedMethodAndTypeProperties
             ? GetMethodInfoUsingManagedNameHelper(testMethod, testClassInfo, discoverInternals)
-            : GetMethodInfoUsingRuntimeMethods(testMethod, testClassInfo, discoverInternals);
+            : throw new UnreachableException();
```

Comment:
Let's use the helper so we can capture some info about where it's thrown if for some reason this was hit.

```suggestion
            : throw ApplicationStateGuard.Unreachable();
```